### DISCUSSION TITLE
Connection to Atlas/MongoDb

### DISCUSSION THREAD

**Discussion by Jovannino** [Budibase/budibase#12457] [Help]
Hello,
I'm trying to connect to a Atlas db without success
This is the link I use with Compass  (using an active AWS Vpn Client).

mongodb://10.198.71.184:27017,10.198.71.184:27018,10.198.71.184:27019/T4H-sample?replicaSet=rs0

Is for Budibase an issue ? 
Thanks
Gio

**Comment by melohagan** [comment]
Are you self hosted @Jovannino ?

**Reply by Jovannino** [reply]
From my PC I connect to AWS using AWS VPN Client and the I'm able to connect to database using Compass. If I try using Budibase it doesn't work . Is it why the connection is using Replica set ? There are 3 server that share the connection . I'm not a tech person but it would be nice to test Budibase against this Atlas db.

**Reply by melohagan** [reply]
Sorry, just to clarify I meant are you self-hosting Budibase? I have tested against Atlas before, using the MongoDB compass URL which has the rough format: `mongodb+srv://...cluster0.mggrb.mongodb.net`

What error are you seeing when you try to connect? My guess at the moment is that there may be something to do with the VPN

**Reply by Jovannino** [reply]
Hi, this is the testing site I'm using now  https://kfive.budibase.app/builder/portal/apps
I  create new source data and then I choose mongodb within the db list. 
I then Insert the uri and db
After a while the error comes out:  Error creating datasource : unable to connect : connection time out.
Using the same VPN after 10 second Compass make the connection to database without any issue and I can check data smoothly.
mongodb://10.198.71.184:27017,10.198.71.184:27018,10.198.71.184:27019/xdemo?replicaSet=rs0&tls=true

If I use your sintax mongodb+srv://10.198.71.184/xdemo?replicaSet=rs0 
Error creating datasource: Unable to connect: querySrv ENOTFOUND _mongodb._tcp.10.198.71.184

**Reply by melohagan** [reply]
I think that's the problem then, you are connecting from Budibase cloud. 

If you have a VPN running on your local machine, Budibase cloud traffic isn't going to route through your PC (and your VPN). 

If you need to use a VPN to connect to a data source, then you'll need to self-host Budibase. 

Furthermore, skimming through this article (https://www.lifewire.com/what-is-a-private-ip-address-2625970) it appears that IP addresses starting with '10' are private, although I don't understand the full implications of that myself.

**Reply by Jovannino** [reply]
Hi,
as said I'm not tech guy but I'm wondering why Compass works smoothly with the same connection and VPN I'm using and Budibase not.
